### ISSUE TITLE
Module to support node_count variable in google_container_node_pool

### ISSUE THREAD

**Issue by Chimbu-DoiT** [terraform-google-modules/terraform-google-kubernetes-engine#1517]
### TL;DR

This module currently not supporting the node_count variable in google_container_node_pool and the values are dynamically calculated based on autoscaling configuration. The issue is that when we update the min_node_count value the cluster autoscaler does not perform scale-up operations since this value is considered only during scale-down activity. 

From https://cloud.google.com/kubernetes-engine/docs/how-to/cluster-autoscaler#node_pool_size_mismatching

```
You may observe a node pool size being smaller than the minimum number of nodes you specified for the cluster. This behaviour happens because the autoscaler uses the minimum number of nodes parameter only when it needs to determine a scaling down.
```

We have to manually update the Number of nodes (per zone) value in the console to trigger the scaling activities; 

![yjobtkyl](https://user-images.githubusercontent.com/112865563/210546619-130df008-5ebd-405f-8b56-8bbc5adf60b9.png)

### Terraform Resources

```markdown
https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_node_pool
```

### Detailed design

```markdown
Allow the module to support the optional node_count variable and use that value in the google_container_node_pool resource if set.
```
